test/over_react/util/react_wrappers_test.dart

```diff
@@ -72,18 +74,49 @@ main() {
           Map clonePropsShallowProps = new Map.from(cloneProps);
           originalShallowProps.remove('internal');
           clonePropsShallowProps.remove('internal');
-          expect(originalShallowProps, equals(clonePropsShallowProps));
+          expect(originalShallowProps, clonePropsShallowProps);
 
           // Verify react-dart internal props are equal.
           ReactDartComponentInternal originalInternal = originalProps['internal'];
           ReactDartComponentInternal clonePropsInternal = cloneProps['internal'];
-          expect(originalInternal.props, equals(clonePropsInternal.props));
+          expect(originalInternal.props, clonePropsInternal.props);
 
           var dartRendered = getDartComponent(render(original));
           var dartRenderedClone = getDartComponent(render(clone));
 
           expect(dartRenderedClone, isNot(same(dartRendered)));
-          expect(dartRenderedClone.props, equals(dartRendered.props));
+          expect(dartRenderedClone.props, dartRendered.props);
+        });
+
+        test('for a Dart component version 2', () {
+          var original = TestComponent2Factory(testPropsForComponent2,
+          testChildren);
+          var clone = cloneElement(original);
+
+          // If these objects are equal, then they proxy the same JS props object.
+          expect(clone.props, isNot(original.props));
+
+          Map originalProps = unconvertJsProps(original);
+          Map cloneProps = unconvertJsProps(clone);
+
+          // Verify all props (including children, excluding react-dart internals) are equal.
+          Map originalShallowProps = new Map.from(originalProps);
+          Map clonePropsShallowProps = new Map.from(cloneProps);
+          originalShallowProps.remove('internal');
+          clonePropsShallowProps.remove('internal');
```

Comment:
> excluding react-dart internals

This doesn't apply to component2, so this part involving `internal` isn't needed